lib/changelog_output.rb

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2013-2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+require_relative "./changelog_source"
+
+class ChangelogOutput
+  attr_reader :output
+
+  def initialize(files)
+    @rpm_changes = files[:rpm_changes]
+    @news = files[:news]
+    @output = ""
+  end
+
+  def rpm_changes
+    rpm_changes = ChangelogSource.new(@rpm_changes)
+    rpm_changes.parse
+    rpm_changes
+  end
+
+  def news
+    news = ChangelogSource.new(@news)
+    news.parse
+    news
+  end
```

Comment:
The `rpm_changes` and `news` method are only used once. It would be simpler, if the code would just be folded back into the methods where they are called, I think.